pyro/distributions/sine_skewed.py

```diff
@@ -0,0 +1,92 @@
+from math import pi
+
+import torch
+from torch.distributions import Uniform
+
+from pyro.distributions import constraints
+
+from .torch_distribution import TorchDistribution
+
+
+class SineSkewed(TorchDistribution):
+    """The Sine Skewed distribution [1] is a distribution for breaking pointwise-symmetry on a base-distribution over
+    the d-dimensional torus.
+
+    This distribution requires the base distribution on a torus. The parameter skewness can be inferred using
+    :class:`~pyro.infer.HMC` or :class:`~pyro.infer.NUTS`. For example, the following will produce a uniform prior
+    over skewness for the 1-torus,::
+
+        def model(...):
+            ...
+            skewness_phi = pyro.sample(f'skewness_phi', Uniform(skewness.abs().sum(), 1 - tots))
+            psi_bound = 1 - skewness_phi.abs()
+            skewness_psi = pyro.sample(f'skewness_psi', Uniform(-psi_bound, psi_bound)
+            skewness = torch.stack((skewness_phi, skewness_psi), dim=0)
+            ...
+
+    .. note:: An event in the base distribution must be on a d-torus, so the event_shape must be (d, 2) or (2,).
```

Comment:
Please add a note for why the last dimention is 2.

Reply:
These are the two angles that specify the position on the torus; I will update the docstring.